client/blocks/utils.js

```diff
@@ -20,14 +20,16 @@ export const getBlocksConfiguration = () => {
  * Determines if off-session payment should be set up.
  *
  * @param {boolean} showSaveOption - Whether to show the save option.
+ * @param {boolean} paymentMethodReusable - Whether the payment method is reusable.
  * @return {boolean} True if off-session payment should be set up, false otherwise.
  */
-export const shouldSetupOffSessionPayment = ( showSaveOption ) => {
-	const config = getBlocksConfiguration();
-	const hasAutoRenewingSubscription =
-		config?.cartContainsSubscription &&
-		! config?.subscriptionRequiresManualRenewal;
-	return hasAutoRenewingSubscription || showSaveOption;
+export const shouldSetupOffSessionPayment = (
+	showSaveOption,
+	paymentMethodReusable
+) => {
+	return (
+		hasAutoRenewingSubscription( paymentMethodReusable ) || showSaveOption
```

Comment:
Non-blocking nit: I think it's easier to read if we check the local boolean before calling the function.
```suggestion
		showSaveOption || hasAutoRenewingSubscription( paymentMethodReusable )
```